lib/dry/schema/value_coercer.rb

```diff
@@ -18,7 +18,7 @@ def call(input)
         if input.success?
           type_schema[Hash(input)]
         else
-          type_schema.member_types.reduce(EMPTY_HASH.dup) do |hash, (name, type)|
+          type_schema.name_key_map.reduce(EMPTY_HASH.dup) do |hash, (name, type)|
```

Comment:
Maybe it should be an enumerable, so that we could do `type_schema.reduce`?

Reply:
yeah, I thought about it, it makes perfect sense now. I'll do this once I finish with testing